pkg/types/plugin.go

```diff
@@ -218,6 +222,23 @@ func (d *pluginDiffer) createUpdatePlugin(plugin *state.Plugin) (*crud.Event, er
 	return nil, nil
 }
 
+func (d *pluginDiffer) getPluginSchema(ctx context.Context, pluginName string) (map[string]interface{}, error) {
+	var schema map[string]interface{}
+
+	d.mu.Lock()
+	defer d.mu.Unlock()
+	if schema, ok := d.schemasCache[pluginName]; ok {
+		return schema, nil
+	}
+
+	schema, err := d.kongClient.Plugins.GetFullSchema(ctx, &pluginName)
+	if err != nil {
+		return schema, err
+	}
+	d.schemasCache[pluginName] = schema
+	return schema, nil
+}
```

Comment:
This looks awfully similar to `Syncer`'s `getPluginSchema`. I think we could write a separate struct which implements the plugins retrieval and cache and reuse that. WDYT?

Reply:
Done.